src/context.ts

```diff
@@ -106,6 +106,21 @@ export class Context {
       });
     }
   }
+  renameFieldInstance(newPath: string, meta: Meta) {
+    const { path, location } = meta;
+    const instance = this.dataMap.get(getDataMapKey(path, location));
+    if (!instance) {
+      throw new Error('Attempt to rename field that did not pre-exist in context');
+    }
+    const { originalPath } = instance;
+    if (this.fields.length !== 1) {
+      throw new Error('Attempt to rename multiple fields.');
+    }
+    if (/\.|\*/g.test(originalPath)) {
+      throw new Error('Attempt to rename a wildcard field');
+    }
```

Comment:
Nice thinking here. I suppose that the callback should be able to handle the name to give to a field though -- as it has access to the full original path.

Reply:
I had some success on renaming wildcard paths but need to do more work for wildcard support